services/state/token/account_airdrop.proto

```diff
@@ -0,0 +1,55 @@
+syntax = "proto3";
+
+package proto;
+
+/*-
+ * ‌
+ * Hedera Network Services Protobuf
+ * ​
+ * Copyright (C) 2018 - 2024 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import "basic_types.proto";
+
+option java_package = "com.hederahashgraph.api.proto.java";
+// <<<pbj.java_package = "com.hedera.hapi.node.state.token">>> This comment is special code for setting PBJ Compiler java package
+option java_multiple_files = true;
+
+
+/**
+ * Representation of a pending airdrops for a given sender account in the network Merkle tree.
+ */
+message AccountAirdrop {
+  /**
+   * The airdrop involved in the relation between a sender account and the airdrop initiated by them
+   */
+  PendingAirdropId pending_airdrop_id = 1;
+
+  /**
+   * The value of the current airdrop id. SHALL NOT be set for non fungible tokens
+   */
+  PendingAirdropValue pending_airdrop_value = 2;
+
```

Comment:
This is purely duplicative.  The pending airdrop ID above can be used to look up the pending airdrop value, and should be.  Duplicating the value here just increases the number of places that need to be updated and increases the chance that we make a mistake.

Reply:
Agreed we don't need the value here.

Reply:
We need this because we are we are changing the state map to be <PendingAirdropId,AccountAirdrop>, do you suggest that we store the `amount` directly as `unit64`

Reply:
Are you saying this will _replace_ the existing `PendingAirdropValue` message in state?
If we're doing that, the HIP needs to change (which should be discussed across teams and include external stakeholders) to include this message, as that is no longer just an "internal" detail.

The existing `PendingAirdropValue` message only exists to hold a single `uint64` value; if we're replacing that message with this one, then we wouldn't need the wrapper and could just store the primitive value here.

Reply:
This changes only the internal representation, which is an implementation detail. The HAPI messages should not derive from the HIP. Why do you think it needs to be discussed across teams? Does the HIP define how the information is stored in state?

Reply:
It does, it describes the storage of pending airdrops as a map from `PendingAirdropId` to `PendingAirdropValue`. We can perhaps debate the wisdom of doing so some other time.

I'm not intending to imply we cannot make a change, only that it probably needs some additional discussion at this stage.

The growth of state for pending airdrops was discussed in the HIP review quite a bit, and there was a significant concern that the storage be as small as possible. Whatever changes are made here should probably keep that in mind.

Reply:
We will look into updating the HIP, as for this value I think we need to use `PendingAirdropValue` and not the `unit64` as we use PendingAirdropValue in the record file as well.

Reply:
It's not a critical point, but there is nothing that requires us to directly match values in state with values in a record file.
As long as records can be created (and we're just passing a Java `long` to the record builder to do that) we can store things in state very differently from the record files.